src/zh/boylove/src/eu/kanade/tachiyomi/extension/zh/boylove/BoyLove.kt

```diff
@@ -121,9 +126,31 @@ class BoyLove : HttpSource(), ConfigurableSource {
                     .filter { it.attr("src").endsWith("load.png") }
                     .map { it.attr("data-original").trim().toImageUrl() }
             }
-            urlList.mapIndexed { index, imageUrl -> Page(index, imageUrl = imageUrl) }
+            val parts = doc.getPartsCount()
+            urlList.mapIndexed { index, imageUrl ->
+                val url = if (parts == null) {
+                    imageUrl
+                } else {
+                    imageUrl.toHttpUrl().newBuilder()
+                        .addQueryParameter("parts", "$parts")
```

Comment:
Make the key a global scope constant in the descrambler file. I'd also use a longer name just in case it conflicts with source url queries.

An alternative is to use fragment/hash so you don't need to handle removing it from URL query.